File: _site/python_solutions/485.max-consecutive-ones.py
#
# @lc app=leetcode id=485 lang=python3
#
# [485] Max Consecutive Ones
#
# https://leetcode.com/problems/max-consecutive-ones/description/
#
# algorithms
# Easy (54.13%)
# Total Accepted:    118.2K
# Total Submissions: 217.8K
# Testcase Example:  '[1,0,1,1,0,1]'
#
# Given a binary array, find the maximum number of consecutive 1s in this
# array.
# 
# Example 1:
# 
# Input: [1,1,0,1,1,1]
# Output: 3
# Explanation: The first two digits or the last three digits are consecutive
# 1s.
# ⁠   The maximum number of consecutive 1s is 3.
# 
# 
# 
# Note:
# 
# The input array will only contain 0 and 1.
# The length of input array is a positive integer and will not exceed 10,000
# 
# 
#
class Solution:
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return max([len(s) for s in ''.join([str(i) for i in nums]).split('0')])

nums = [1,1,0,1,1,1]
nums = [0]
print(Solution().findMaxConsecutiveOnes(nums))

File: _site/solutions/485_max_consecutive_ones.cpp
#include<stdio.h>
#include<map>
#include<iostream>
#include<vector>
#include<algorithm>
#include<climits>
/**
   Given a binary array, find the maximum number of consecutive 1s in this array.
   Example 1:
   Input: [1,1,0,1,1,1]
   Output: 3
   Explanation: The first two digits or the last three digits are consecutive 1s.
   The maximum number of consecutive 1s is 3.
   Note:
   The input array will only contain 0 and 1.
   The length of input array is a positive integer and will not exceed 10,000

**/
using namespace std; 

static int speed_up = []() { std::ios::sync_with_stdio(false); cin.tie(NULL); return 0; }(); 

class Solution {
public:
  int findMaxConsecutiveOnes(vector<int>& nums) {
    if (nums.empty()) return 0;
    int ans = 0, aux = 0;
    for(n:nums) 
      if (n==1){
	aux ++;
	ans = max(ans, aux); 
      } else {
	aux = 0; 
      }
    return ans; 
  }
};

int main() { 
  Solution s; 
}
